orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/cluster/FindImageFromClusterTask.groovy

```diff
@@ -173,30 +176,19 @@ class FindImageFromClusterTask extends AbstractCloudProviderAwareTask implements
         throw new IllegalStateException("Missing image on ${deploymentDetailTemplate}")
       }
 
-      def mkDeploymentDetail = { String imageName, String imageId ->
-        [
-          imageId        : imageId,
-          imageName      : imageName,
-          serverGroupName: config.cluster,
-          image          : deploymentDetailTemplate.image + [imageId: imageId, name: imageName],
-          buildInfo      : deploymentDetailTemplate.buildInfo ?: [:]
-        ]
-      }
-
       List<Map> images = oortService.findImage(cloudProvider, searchNames[0] + '*', account, null, null)
-      for (Map image : images) {
-        for (Location location : missingLocations) {
-          if (imageSummaries[location] == null && image.amis && image.amis[location.value]) {
-            imageSummaries[location] = [
-              mkDeploymentDetail((String) image.imageName, (String) image.amis[location.value][0])
-            ]
-          }
-        }
-      }
+      resolveFromBaseImageName(images, missingLocations, imageSummaries, deploymentDetailTemplate, config)
 
       def unresolved = imageSummaries.findResults { it.value == null ? it.key : null }
       if (unresolved) {
-        throw new IllegalStateException("Still missing images in $unresolved.value")
+        // fallback to look it default bake account; the deploy operation will execute the allowLaunchOperation to share
+        // the image into the target account
+        List<Map> defaultImages = oortService.findImage(cloudProvider, searchNames[0] + '*', defaultBakeAccount, null, null)
```

Comment:
I think this fallback lookup should only be performed if `cloudProvider == 'aws'` (or maybe 'titus' too, not sure). But doesn't really apply for gce.